lib/WaterlineLifecycleCallback.js

```diff
@@ -26,6 +23,33 @@ var self = {
         modelToChange[method.method] = newFunction;
       });
     }
+  },
+
+  insertHistoryRecord: function(data, actionType, historyModel){
+    data.actionType = actionType;
+    global[historyModel].create(data).exec(function createHistory(err, model) {
+      if (err) {
+        sails.log.debug('Error occurred in saving history ', err);
+      } else {
+        sails.log.debug('History record created!');
+      }
+    });
+  },
+
+  insertHistoryDeletedRecord: function(deletedRecords, actionType, historyModel, fieldAttributes){
+    _.each(deletedRecords, function eachInstantiatedModel(data, identity) {
+
+      self.insertHistoryRecord(self.transformToSailsModel(data, fieldAttributes), actionType, historyModel);
+    });
+  },
+
+  transformToSailsModel: function(data, fieldAttributes){
+    var transformedData = {};
+    _(fieldAttributes).forEach(function buildSubsetOfAssociations(attrDef, attrName) {
```

Comment:
Instead of iterating the fieldAttributes, should've implemented a reverse lookup wherein the key is already the `columnName` and the value is the property. Please modify the approach.